libkernelflinger/ivshmem.c

```diff
@@ -157,7 +157,22 @@ typedef union {
 	UINT32 uint32;
 } __attribute__((packed))  pci_config_address_t;
 
-struct ivshmem_device g_ivshmem_dev;
+struct ivshmem_device {
+	UINT8 dev;
+	UINT8 func;
+	UINT8 revision;
+
+	UINT32 bar0_addr;
+	UINT32 bar0_len;
+	UINT32 bar1_addr;
+	UINT32 bar1_len;
+	UINT32 bar2_addr;
+	UINT32 bar2_len;
+};
+
+static struct ivshmem_device g_ivshmem_dev;
+
+UINT64 g_ivshmem_rot_addr = 0;
```

Comment:
no need to add this global variable

Reply:
How do you put rot in ivshmem without this global variable?

Reply:
you already defined g_ivshmem_dev, it can use different OFFSET for rot and rollback index handing